activerecord/test/cases/relations_test.rb

```diff
@@ -942,6 +942,50 @@ def test_destroy_all_with_conditions_is_deprecated
     end
   end
 
+  def test_destroy_all_deprecated_return_value
+    ActiveRecord::Base.destroy_all_in_batches = false
+
+    davids = Author.where(:name => 'David')
+
+    assert_deprecated do
+      assert_difference('Author.count', -1) do
+        assert_equal davids.to_a, davids.destroy_all
+      end
+    end
+  ensure
+    ActiveRecord::Base.destroy_all_in_batches = true
+  end
+
+  def test_destroy_all_in_batches
+    davids = Author.where(:name => 'David')
+
+    assert_difference('Author.count', -1) do
+      assert_nil davids.destroy_all
+    end
+  end
+
+  def test_destroy_all_in_batches_returns_nothing
+    davids = Author.where(:name => 'David')
+
+    assert_difference('Author.count', -1) do
+      assert_nil davids.destroy_all
+    end
+  end
+
+  def test_destroy_all_in_batches_delegates_work_to_in_batches
+    davids = Author.where(:name => 'David')
+
+    assert_called(davids, :in_batches, times: 2, returns: davids.in_batches) do
+      davids.destroy_all
+    end
+  end
+
+  def test_destroy_all_in_batches_ignores_limits
```

Comment:
I pushed it this afternoon https://github.com/rails/rails/commit/451437c6f57e66cc7586ec966e530493927098c7.